src/Hub.php

```diff
@@ -83,31 +87,63 @@ public function getFactory(): ?TokenFactoryInterface
      */
     public function publish(Update $update): string
     {
-        $postData = [
-            'topic' => $update->getTopics(),
-            'data' => $update->getData(),
-            'private' => $update->isPrivate() ? 'on' : null,
-            'id' => $update->getId(),
-            'type' => $update->getType(),
-            'retry' => $update->getRetry(),
-        ];
-
         $jwt = $this->getProvider()->getJwt();
         $this->validateJwt($jwt);
+        $postData = $this->updateToArray($update);
 
         try {
-            return $this->httpClient->request('POST', $this->getUrl(), [
+            $response = $this->httpClient->request('POST', $this->getUrl(), [
                 'auth_bearer' => $jwt,
                 'headers' => [
                     'Content-Type' => 'application/x-www-form-urlencoded',
                 ],
                 'body' => Internal\QueryBuilder::build($postData),
             ])->getContent();
+
+            $this->logUpdate($update);
+
+            return $response;
         } catch (ExceptionInterface $exception) {
             throw new Exception\RuntimeException('Failed to send an update.', 0, $exception);
         }
     }
 
+    private function updateToArray(Update $update): array
+    {
+        return [
+            'topic' => $update->getTopics(),
+            'data' => $update->getData(),
+            'private' => $update->isPrivate() ? 'on' : null,
+            'id' => $update->getId(),
+            'type' => $update->getType(),
+            'retry' => $update->getRetry(),
+        ];
+    }
+
+    private function logUpdate(Update $update): void
+    {
+        if (null === $this->logger) {
+            return;
+        }
+
+        $this->logger->info(
+            sprintf(
+                'Published update with ID \'%s\' on topic(s) [%s]',
+                $update->getId(),
+                implode(', ', $update->getTopics())
+            ),
+            [
+                'id' => $update->getId(),
+                'topic' => $update->getTopics(),
+            ]
+        );
+
+        $this->logger->debug(
+            sprintf('Debug details for update \'%s\'', $update->getId()),
+            $this->updateToArray($update)
+        );
+    }
```

Comment:
IMHO it would be better to have only one entry in the logs instead of two, even in debug mode. What do you think about like that:

```suggestion
        // kernel.debug parameter
        $this->logger->info(
            sprintf(
                'Published update with ID "%s" on topic(s) [%s]',
                $update->getId(),
                implode(', ', $update->getTopics())
            ),
            $this->debug ?  $this->updateToArray($update) : [
                'id' => $update->getId(),
                'topic' => $update->getTopics(),
            ] : 
        );
    }
```

`$this->debug` will be a new promoted property defaulting to false in the component, and set to `%kernel.debug%` by the bundle.

This approach would also save 2 function calls in non-debug mode.